code/game/mecha/combat/marauder.dm

```diff
@@ -162,78 +161,107 @@
+		var/obj/mecha/combat/marauder/Marauder = linked_mech
+		if(Marauder.lock_controls)
+			return
+		if(Marauder.occupant)
+			if(Marauder.get_charge() <= 0)
+				return
 
-			var/landing = get_distant_turf(get_turf(src), dir, 5)
-			throw_at(landing, 5 , 2)
+		Marauder.crashing = null
+		var/landing = get_distant_turf(get_turf(linked_mech), Marauder.dir, 5)
+		Marauder.throw_at(landing, 5 , 2)
 
-			dash_ready = 0
-			spawn(dash_cooldown)
-				dash_ready = 1
-			src.log_message("Performed Rocket-Dash.")
-			src.occupant_message("Triggered Rocket-Dash sub-routine")
+		Marauder.log_message("Performed Rocket-Dash.")
+		Marauder.occupant_message("Triggered Rocket-Dash sub-routine")
 	return
 
-/obj/mecha/combat/marauder/verb/smoke()
-	set category = "Exosuit Interface"
-	set name = "Smoke"
-	set src = usr.loc
-	set popup_menu = 0
-	if(usr!=src.occupant)
+
+/spell/mech/marauder/smoke
+	name = "Smoke"
+	desc = "Deploy obscuring smoke to avoid retaliation."
+	charge_max = 100
+	charge_counter = 100
+	override_icon = 'icons/mob/screen_spells.dmi'
+	hud_state = "wiz_smoke"
+
+/spell/mech/marauder/smoke/cast(list/targets, mob/user)
+	if(user!=linked_mech.occupant)
 		return
-	if(smoke_ready && smoke>0)
-		src.smoke_system.start()
-		smoke--
-		smoke_ready = 0
-		spawn(smoke_cooldown)
-			smoke_ready = 1
+	var/obj/mecha/combat/marauder/Marauder = linked_mech
+	if(Marauder.smoke>0)
+		Marauder.smoke_system.start()
+		Marauder.smoke--
 	return
 
-/obj/mecha/combat/marauder/verb/zoom()
-	set category = "Exosuit Interface"
-	set name = "Zoom"
-	set src = usr.loc
-	set popup_menu = 0
-	if(usr!=src.occupant)
+/spell/mech/marauder/zoom
+	name = "Zoom"
+	desc = "Double your viewing distance."
+	override_icon = 'icons/obj/items.dmi'
+	hud_state = "binoculars"
+
+/spell/mech/marauder/zoom/cast(list/targets, mob/user)
+	if(user!=linked_mech.occupant)
 		return
-	if(src.occupant.client)
-		src.zoom = !src.zoom
-		src.log_message("Toggled zoom mode.")
-		src.occupant_message("<font color='[src.zoom?"blue":"red"]'>Zoom mode [zoom?"en":"dis"]abled.</font>")
-		if(zoom)
-			src.occupant.client.changeView(12)
-			src.occupant << sound('sound/mecha/imag_enh.ogg',volume=50)
+	if(linked_mech.occupant.client)
+		var/obj/mecha/combat/marauder/Marauder = linked_mech
+		Marauder.zoom = !Marauder.zoom
+		Marauder.log_message("Toggled zoom mode.")
+		Marauder.occupant_message("<font color='[Marauder.zoom?"blue":"red"]'>Zoom mode [Marauder.zoom?"en":"dis"]abled.</font>")
+		if(Marauder.zoom)
+			Marauder.occupant.client.changeView(12)
+			Marauder.occupant << sound('sound/mecha/imag_enh.ogg',volume=50)
 		else
-			src.occupant.client.changeView()//world.view - default mob view size
+			Marauder.occupant.client.changeView()//world.view - default mob view size
 	return
 
 
+/obj/mecha/combat/marauder/refresh_spells()
+	if(!occupant)
+		return
+	occupant.add_spell(new /spell/mech/marauder/thrusters(src), "mech_spell_ready", /obj/abstract/screen/movable/spell_master/mech)
```

Comment:
You can have a list of spells intrinsic to the mech and loop through it instead of giving it each of the spells.